### PR TITLE
fix(security-rules): unblock v3 categories

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

_What this PR does_

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -82,6 +82,11 @@ service cloud.firestore {
       allow write: if isPublicWritable();
     }
 
+    match /v3_categories/{document=**} {
+      allow read: if isPublicReadable();
+      allow write: if isPublicWritable();
```
Comment (mariojsnunes) [inline firestore.rules:87]: the write should at least be restrict by authenticated user.

note that currently it's not possible to create categories via our UI.
but unsure how to make a rule here that would deny via our UI but allow wherever they are created now (maybe retool?)

**Review by mariojsnunes — CHANGES_REQUESTED** [review]

File: firestore.rules (hunk 1 above)
Comment (mariojsnunes) [reply firestore.rules:87]: and the same for
question_categories_rev20231130, research_categories_rev20221224, v3_tags

**Comment by mariojsnunes** [comment]
to fix all the erros:
```
match /v3_categories/{document=**} {
  allow read: if isPublicReadable();
  allow write: if request.auth != null;
}
    
match /_archived/v3_categories/{document=**} {
  allow read: if isPublicReadable();
  allow write: if request.auth != null;
}

match /_archived/research_categories_rev20221224/{document=**} {
  allow read: if isPublicReadable();
  allow write: if request.auth != null;
}

match /_archived/question_categories_rev20231130/{document=**} {
  allow read: if isPublicReadable();
  allow write: if request.auth != null;
}

match /_archived/v3_tags/{document=**} {
  allow read: if isPublicReadable();
  allow write: if request.auth != null;
}
```

File: firestore.rules (hunk 1 above)
Comment (thisislawatts) [reply firestore.rules:87]: My current aim is to introduce rules for each collection used in our app, that aligns with the previous wide open permissions. From this baseline we can then iterate over each collection, introducing permissions that better match the intended access patterns. 

I want to make these changes in an isolated way. The aim here is that we can easily isolate side affects and roll back as necessary. 

There will be a follow up PR that locks down these 'admin-only' category and tag collections. These documents should only be modifiable by moderators using Retool, so I think we can entirely block write access in firestore.rules.

Retool uses a service account for auth when accessing the database which bypasses these controls.

**Pull request merged**